torch/distributed/device_mesh.py

```diff
@@ -411,31 +411,47 @@ def __getitem__(
             self, mesh_dim_names: Union[str, Tuple[str, ...]]
         ) -> "DeviceMesh":
             """
-            Slice the current DeviceMesh based on the mesh_dim_name given to create a child
-            DeviceMesh.
+            Slice the current DeviceMesh based on the mesh_dim_names given to create a child
+            DeviceMesh. This supports 1D or nD slicing. The DeviceMesh created will have dimensions
+            corresponding to the length of the provided mesh_dim_names and the order of dimensions
+            will be consistent with the order in which the mesh_dim_names were provided.
 
             Args:
-                mesh_dim_name (Union[str, Tuple[str]]): the name or the tuple of names of the
+                mesh_dim_names (Union[str, Tuple[str]]): the name or the tuple of names of the
                 mesh dimension of the parent DeviceMesh to create the child DeviceMesh for.
             Returns:
                 A :class:`DeviceMesh` object
 
-            The following program runs on each process/rank in an SPMD manner. In this example, we have 2
-            hosts with 4 GPUs each.
-            Calling mesh["tp"] on rank 0, 1, 2, 3 would return a 1D child DeviceMesh:([0, 1, 2, 3]).
-            Calling mesh["tp"] on rank 4, 5, 6, 7 would return a 1D child DeviceMesh:([4, 5, 6, 7]).
-            Calling mesh["dp"] on rank 0, 4 would return a 1D child DeviceMesh:([0, 4]).
-            Calling mesh["dp"] on rank 1, 5 would return a 1D child DeviceMesh:([1, 5]).
-            Calling mesh["dp"] on rank 2, 6 would return a 1D child DeviceMesh:([2, 6]).
-            Calling mesh["dp"] on rank 3, 7 would return a 1D child DeviceMesh:([3, 7]).
+            The following program runs on each process/rank in an SPMD manner in a world size of 8.
+            In the first example:
+                Calling mesh_2d["tp"] on rank 0, 1, 2, 3 returns a 1D child DeviceMesh:([0, 1, 2, 3]).
```

Comment:
I think we should change the wording from "child DeviceMesh" to sth like "sub DeviceMesh"? A child indicates a parent somewhere, where we should avoid using this term going forward. Please rename all the comments

Reply:
Removed "child DeviceMesh" and replaced with submesh instead. I am not sure if we want to expose the concept of root mesh yet, as root mesh only lives in our mesh env right now.